tests/system/src/test/java/test/com/sun/javafx/animation/AnimationTimerTest.java

```diff
@@ -25,88 +25,36 @@
+import org.junit.jupiter.api.Test;
+
 import javafx.animation.AnimationTimer;
-import javafx.application.Application;
-import javafx.application.Platform;
-import javafx.stage.Stage;
-import org.junit.AfterClass;
-import org.junit.BeforeClass;
-import org.junit.Test;
 import test.util.Util;
 
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.junit.jupiter.api.Assertions.assertThrows;
-
-public class AnimationTimerTest {
-
-    private static final CountDownLatch startupLatch = new CountDownLatch(1);
-
-    private static Stage primaryStage;
-
-    public static class TestApp extends Application {
-
-        @Override
-        public void init() throws Exception {
-            assertFalse(Platform.isFxApplicationThread());
-        }
-
-        @Override
-        public void start(Stage stage) throws Exception {
-            primaryStage = stage;
-            assertTrue(Platform.isFxApplicationThread());
-
-            startupLatch.countDown();
-        }
-
-    }
-
-    @BeforeClass
-    public static void setup() throws Exception {
-        Util.launch(startupLatch, TestApp.class);
-    }
-
-    @AfterClass
-    public static void shutdown() {
-        Util.shutdown(primaryStage);
-    }
+public class AnimationTimerTest extends SynchronizationTest {
 
     @Test
-    public void animationTimerOnFXThreadTest() throws InterruptedException {
-        final CountDownLatch frameCounter = new CountDownLatch(3);
-        Platform.runLater(() -> {
-            AnimationTimer timer = new AnimationTimer() {
-                @Override public void handle(long l) {
-                    frameCounter.countDown();
-                    if (frameCounter.getCount() == 0L) {
-                        stop();
-                    }
-                }
-            };
-            assertTrue(Platform.isFxApplicationThread());
-            timer.start();
-        });
-        frameCounter.await();
+    public void testAnimationTimer() throws InterruptedException {
+        runTest(this::startAnimationTimer);
     }
 
-    @Test
-    public void startAnimationTimerNotOnFXThreadTest() {
-        assertFalse(Platform.isFxApplicationThread());
-        AnimationTimer timer = new AnimationTimer() {
-            @Override public void handle(long l) {}
-        };
-        assertThrows(IllegalStateException.class, timer::start);
-    }
+    private void startAnimationTimer() {
+        registerExceptionHandler();
 
-    @Test
-    public void stopAnimationTimerNotOnFXThreadTest() {
-        assertFalse(Platform.isFxApplicationThread());
-        AnimationTimer timer = new AnimationTimer() {
-            @Override public void handle(long l) {
-                assertThrows(IllegalStateException.class, () -> stop());
+        var timer = new AnimationTimer() {
+
+            @Override
+            public void handle(long now) {
+                // Simulate intensive processing
+                Util.sleep(10);
```

Comment:
Would this be more likely to catch a problem if this sleep (which is on the FX thread) were reduced to 1 msec? Not sure, but since the timer is started and stopped every 10 msec, it might be worth looking at. On the other hand, it does detect the bug on my Mac, so it might be fine as is.